src/types/InterfaceType.ts

```diff
@@ -58,13 +58,12 @@ export class InterfaceType implements BscType {
         return 'object';
     }
 
-    public equals(targetType: BscType): boolean {
+    public equals(targetType: BscType, context?: TypeContext): boolean {
+        let targetTypeString = targetType?.toString(context);
         if (isInterfaceType(targetType)) {
-            if (targetType.members.size !== this.members.size) {
-                return false;
-            }
-            return targetType.isAssignableTo(this);
+            targetTypeString = targetType.toJsString();
         }
-        return false;
+        return this.toJsString() === targetTypeString;
```

Comment:
Why are we comparing a string rather than looping over the objects themselves? 

Also, you might need to sort the keys before turning into json. otherwise these two interfaces won't be equivalent. 
```vb
interface Thing1
    name as string
    age as integer
end interface

interface Thing2
    age as integer
    name as string
end interface
```

Reply:
> Why are we comparing a string rather than looping over the objects themselves?

cause this was easier .... 

I'll fix it. :)

Reply:
Also consider #517 and #516 , I think the string version probably won't allow special handling for those types.